### PR TITLE
Scaffold the Rails application

### PR DESCRIPTION
**Pull request by AmberWilkie**
PT Story: https://www.pivotaltracker.com/story/show/131070665

Changes proposed in this pull request:
- Scaffolds new Rails 5 application
- Adds Rspec and Cucumber
- Sets up Travis and Coveralls continuous integration

What I have learned working on this feature: 
??

Ready for review!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+before_script:
+- bundle exec rake db:create --all
+- bundle exec rake db:migrate
+script:
+- bundle exec rake ci:tests
+services:
+- postgresql
+# deploy:
+  # provider: heroku
+  # app: garden-club
```
Comment (tochman) [inline .travis.yml]: ??

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  gem 'coveralls', require: false
+end
+
+group :development do
+  # Access an IRB console on exception pages or by using <%= console %> anywhere in the code.
+  gem 'web-console'
+  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
+  gem 'spring'
+  gem 'spring-watcher-listen', '~> 2.0.0'
+end
```
Comment (tochman) [inline Gemfile:28]: Please clean up comments from this file

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# Use postgresql as the database for Active Record
+gem 'pg', '~> 0.18'
+# Use Puma as the app server
+gem 'puma', '~> 3.0'
+# Use SCSS for stylesheets
+gem 'sass-rails', '~> 5.0'
+# Use Uglifier as compressor for JavaScript assets
+gem 'uglifier', '>= 1.3.0'
+# Use CoffeeScript for .coffee assets and views
+gem 'coffee-rails', '~> 4.2'
```
Comment (tochman) [inline Gemfile]: Are you planning on using coffee?

File: app/assets/javascripts/cable.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+//= require action_cable
+//= require_self
+//= require_tree ./channels
+
+(function() {
+  this.App || (this.App = {});
+
+  App.cable = ActionCable.createConsumer();
+
+}).call(this);
```
Comment (tochman) [inline app/assets/javascripts/cable.js]: Will you be using ActionCable?

File: app/channels/application_cable/channel.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Channel < ActionCable::Channel::Base
+  end
+end
```
Comment (tochman) [inline app/channels/application_cable/channel.rb]: Will you be using ActionCable? See also below.

File: app/mailers/application_mailer.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: 'from@example.com'
+  layout 'mailer'
+end
```
Comment (tochman) [inline app/mailers/application_mailer.rb:4]: Will you be using ActionMailer?

File: config/database.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+#     url: <%= ENV['DATABASE_URL'] %>
+#
+production:
+  <<: *default
+  database: slow_food_production
+  username: slow_food
+  password: <%= ENV['SLOW_FOOD_DATABASE_PASSWORD'] %>
+
+cucumber:
+  <<: *test
```
Comment (tochman) [inline config/database.yml:21]: Please clear out comments from this file

**Review by tochman — CHANGES_REQUESTED** [review]
First review. Plz address the issues. /T

File: Gemfile (hunk 3 above)
Comment (AmberWilkie) [reply Gemfile]: Is there a way to remove these things when we generate Rails? The default builds them all.

File: .travis.yml (hunk 1 above)
Comment (AmberWilkie) [reply .travis.yml]: We will be uncommenting this later when we establish continuous deployment.

File: Gemfile (hunk 3 above)
Comment (tochman) [reply Gemfile]: I know. These decision must be made after scaffold

File: .travis.yml (hunk 1 above)
Comment (tochman) [reply .travis.yml]: Then you should add it at that point. Let's try to keep the code as clean as possible to make the review process simple.

File: app/mailers/application_mailer.rb (hunk 6 above)
Comment (AmberWilkie) [reply app/mailers/application_mailer.rb:4]: Yes we will eventually be using this to email confirms and send information to Restaurants.

Comment (tochman) [reply app/mailers/application_mailer.rb:4]: Okay, thx

**Comment by tochman** [comment]
Please include a link to the PT feature/chore in the PR description as well as what acceptance criteria it delivers.

**Comment by tochman** [comment]
I assume that this deals with https://www.pivotaltracker.com/story/show/131070665
Plz see my comments.

File: config/database.yml (hunk 7 above)
Comment (AmberWilkie) [reply config/database.yml:21]: done

File: Gemfile (hunk 2 above)
Comment (AmberWilkie) [reply Gemfile:28]: done

File: Gemfile (hunk 3 above)
Comment (AmberWilkie) [reply Gemfile]: we have removed this gem

File: .travis.yml (hunk 1 above)
Comment (AmberWilkie) [reply .travis.yml]: alright we have deleted it

**Comment by tochman** [comment]
Here is a good template for Pull Requests that will allow a reviewer to get a sense what the PR is about. 

```
PT Story: [paste pivotal tracker link here]

Changes proposed in this pull request:

[item 1 - replace me]
[item 2 - replace me]
...
...
What I have learned working on this feature: [If you don't put anything here you are doing it wrong!]

Screenshots: [If you made some visual changes to the application please upload screenshots here, or remove this section]

Ready for review!
```

This template can be added to your repository as `.github/PULL_REQUEST_TEMPLATE.md` and will automatically populate the description field once in place. 
Also, please in the future let's keep a PR to one Chore or Feature only.  I notice that this one actually deals with more than one chore. Not a blocker but worth thinking about.

File: app/channels/application_cable/channel.rb (hunk 5 above)
Comment (AmberWilkie) [reply app/channels/application_cable/channel.rb]: no. we have removed these files and project references to ActionCable

**Review by tochman — APPROVED** [review]

**Pull request merged**